src/leiningen/ring/uberwar.clj

```diff
@@ -13,26 +17,48 @@
     (get-cp project)
     (->> (:library-path project) io/file .listFiles (map str))))
 
-(defn contains-entry? [^java.io.File file ^String entry]
-  (with-open [jar-file (JarFile. file)]
-    (some (partial = entry)
-          (map #(.getName ^JarEntry %)
-               (enumeration-seq (.entries jar-file))))))
-
 (defn jar-dependencies [project]
   (for [pathname (get-classpath project)
-        :let [file (io/file pathname)
-              fname (.getName file)]
-        :when (and (.endsWith fname ".jar")
-                   ;; Servlet container will have it's own servlet-api impl
-                   (not (contains-entry? file "javax/servlet/Servlet.class")))]
-    file))
-
-(defn jar-entries [war project]
-  (doseq [jar-file (jar-dependencies project)]
-    (let [dir-path (.getParent jar-file)
-          war-path (war/in-war-path "WEB-INF/lib/" dir-path jar-file)]
-      (war/file-entry war project war-path jar-file))))
+        :when (str/ends-with? pathname ".jar")]
+    (io/file pathname)))
+
+(defn- contains-javax-servlet-class? [^JarFile jar-file]
+  (some? (.getEntry jar-file "javax/servlet/Servlet.class")))
+
+(defn- pom-properties? [^JarEntry entry]
```

Comment:
This is not a predicate, so the name shouldn't end with a `?`. It's probably better to split this into two functions:

```clojure
(defn- pom-properties? [^JarEntry entry]
  (str/ends-with? (.getName entry) "pom.properties"))

(defn- find-pom-properties [^JarFile jar-file]
  (->> jar-file .entries enumeration-seq (filter pom-properties?) first))
```

Reply:
`(first (filter ...))` is kind of an anti-pattern... that's why we have `some` right?